cpp/test/test_impedance_spectroscopy.cc

```diff
@@ -23,148 +23,158 @@
+    };
+  }
+  else
+  {
+    throw std::runtime_error("invalid device type " + device_type);
+  }
 }
 
-void scan(std::shared_ptr<cap::EnergyStorageDevice> dev, std::shared_ptr<boost::property_tree::ptree const> dev_database,
-    std::shared_ptr<boost::property_tree::ptree const> eis_database, std::ostream & os = std::cout)
+void scan(std::shared_ptr<cap::EnergyStorageDevice> dev,
+          std::shared_ptr<boost::property_tree::ptree const> dev_database,
+          std::shared_ptr<boost::property_tree::ptree const> eis_database,
+          std::ostream &os = std::cout)
 {
-    double const frequency_upper_limit = eis_database->get<double>("frequency_upper_limit");
-    double const frequency_lower_limit = eis_database->get<double>("frequency_lower_limit");
-    int    const steps_per_decade      = eis_database->get<int   >("steps_per_decade"     );
-    double const pi = boost::math::constants::pi<double>();
-    std::vector<int> const harmonics = cap::to_vector<int>(eis_database->get<std::string>("harmonics"));
-    double const percent_tolerance = eis_database->get<double>("percent_tolerance");
-
-
-    auto compute_exact = get_compute_exact(dev_database, eis_database);
-
-    double expe_frequency;
-    std::complex<double> expe_impedance;
-    double theo_frequency;
-    std::complex<double> theo_impedance;
-
-    std::shared_ptr<boost::property_tree::ptree> tmp =
-        std::make_shared<boost::property_tree::ptree>(*eis_database);
-
-    os<<"# impedance Z(f) = R + i X \n";
-    os<<boost::format( "# %22s  %22s  %22s  %22s  %22s  \n")
-        % "frequency_f_[Hz]"
-        % "resistance_R_[ohm]"
-        % "reactance_X_[ohm]"
-        % "magnitude_|Z|_[ohm]"
-        % "phase_arg(Z)_[degree]"
-        ;
-
-    for (double frequency = frequency_upper_limit; frequency >= frequency_lower_limit; frequency /= std::pow(10.0, 1.0/steps_per_decade))
+  double const frequency_upper_limit =
+      eis_database->get<double>("frequency_upper_limit");
+  double const frequency_lower_limit =
+      eis_database->get<double>("frequency_lower_limit");
+  int const steps_per_decade = eis_database->get<int>("steps_per_decade");
+  double const pi = boost::math::constants::pi<double>();
+  std::vector<int> const harmonics =
+      cap::to_vector<int>(eis_database->get<std::string>("harmonics"));
+  double const percent_tolerance =
+      eis_database->get<double>("percent_tolerance");
+
+  auto compute_exact = get_compute_exact(dev_database, eis_database);
+
+  double expe_frequency;
+  std::complex<double> expe_impedance;
+  double theo_frequency;
+  std::complex<double> theo_impedance;
+
+  std::shared_ptr<boost::property_tree::ptree> tmp =
+      std::make_shared<boost::property_tree::ptree>(*eis_database);
+
+  os << "# impedance Z(f) = R + i X \n";
+  os << boost::format("# %22s  %22s  %22s  %22s  %22s  \n") %
+            "frequency_f_[Hz]" % "resistance_R_[ohm]" % "reactance_X_[ohm]" %
+            "magnitude_|Z|_[ohm]" % "phase_arg(Z)_[degree]";
```

Comment:
We don't have to use a stupid machine to indent everything. If you want, I can send every files and you can do the indentation by hand ;-)